src/emqtt.erl

```diff
@@ -2227,7 +2220,8 @@ process_incoming(Bytes, Packets, State = #state{parse_state = ParseState, socket
         {more, NParseState} ->
             {keep_state, State#state{parse_state = NParseState}, next_events(Via, Packets)}
     catch
-        error:Error:Stacktrace -> {stop, {Error, Stacktrace}}
+        error:Reason:St ->
+            maybe_shutdown({parse_packets_error, Reason, St}, State)
```

Comment:
I don't think this would be the case for reconnect.  It is the server side violating the protocol isn't it?

Reply:
Yes, but if this happens as a client we need to reconnect (because the user have set the reconnect option), maybe the problem will be fixed after reconnected.